### PR TITLE
DCK-51 Added attendee paragraph with functionality to set attende names.

### PR DESCRIPTION
**Pull request by andriyun**
## Issue

> **PLEASE, PROVIDE SHORTLY WHAT YOU HAVE MADE AT THIS PR**

## Steps for review:

- [ ] Buy ticket
- [ ] Go to My ticket tab on `/user` page
- [ ] Test all possible way to add user information.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer.
- [ ] All tests are running and there are no failed tests reported by CI.
- [ ] Documentation has been updated according to PR changes.
- [ ] Steps for review have been provided according to PR changes.

Thank you for your сontribution and help to Drupal Ukraine!

### CODE REVIEW

File: docroot/modules/custom/dckyiv_commerce/src/Controller/DckyivCommerceController.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   * @throws
+   *
+   * @return array
+   *   The attendee form.
+   */
+  public function attendeeFormAdd(UserInterface $user, OrderItemInterface $commerce_order_item) {
+    /** @var ParagraphInterface $attendee_paragraph */
+    $attendee_paragraph = Paragraph::create(['type' => 'attendee']);
+    $attendee_paragraph->isNew();
+    $form = \Drupal::service('entity.form_builder')->getForm($attendee_paragraph, 'default');
```
[the remark this answers is not in the document]
Comment (andriyun) [reply docroot/modules/custom/dckyiv_commerce/src/Controller/DckyivCommerceController.php]: Fixed

File: docroot/modules/custom/dckyiv_commerce/src/Controller/DckyivCommerceController.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+   *
+   * @return array
+   *   The attendee form.
+   */
+  public function attendeeFormEdit(UserInterface $user, OrderItemInterface $commerce_order_item, ParagraphInterface $attendee_paragraph) {
+    if ($commerce_order_item != $attendee_paragraph->getParentEntity()->id()
+      || $commerce_order_item->getOrder()->getCustomerId() != $user->id()) {
+      throw new AccessDeniedHttpException();
+    }
+    $form = \Drupal::service('entity.form_builder')->getForm($attendee_paragraph, 'edit');
```
[the remark this answers is not in the document]
Comment (andriyun) [reply docroot/modules/custom/dckyiv_commerce/src/Controller/DckyivCommerceController.php]: Fixed

**Comment by andriyun** [comment]
@ivan-berezhnov free to review this PR :)

File: docroot/modules/custom/dckyiv_commerce/src/Plugin/Field/FieldFormatter/AttendeeFormatter.php [hunk 3]
```diff
@@ -84,14 +88,18 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
           '#attributes' => ['class' => ['attendee--wrapper']],
           'attendee' => [
             '#type' => 'link',
-            '#title' => $this->t('Add attendee info'),
+            '#title' => $this->t('Додати'),
```
Comment (anpolimus) [inline docroot/modules/custom/dckyiv_commerce/src/Plugin/Field/FieldFormatter/AttendeeFormatter.php]: English should be basic language.

File: docroot/modules/custom/dckyiv_commerce/dckyiv_commerce.routing.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ dckyiv_commerce.attendee_edit:
   path: '/user/{user}/my-tickets/{commerce_order_item}/attendee/{attendee_paragraph}'
   defaults:
     _controller: '\Drupal\dckyiv_commerce\Controller\DckyivCommerceController::attendeeFormEdit'
-    _title: 'Edit attendee'
+    _title: 'Iнформація про учасника'
```
Comment (anpolimus) [inline docroot/modules/custom/dckyiv_commerce/dckyiv_commerce.routing.yml]: Basic language is English.

File: docroot/modules/custom/dckyiv_commerce/src/Plugin/Field/FieldFormatter/AttendeeFormatter.php [hunk 5]
```diff
@@ -66,15 +66,19 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
         $elements[$delta]['edit-attendee'] = [
           '#type' => 'link',
           '#weight' => 10,
-          '#title' => $empty ? $this->t('Add info') : $this->t('Edit'),
+          '#title' => $empty ? $this->t('Додати інфо') : $this->t('Редагувати'),
```
Comment (anpolimus) [inline docroot/modules/custom/dckyiv_commerce/src/Plugin/Field/FieldFormatter/AttendeeFormatter.php]: Basic language is English

File: docroot/modules/custom/dckyiv_commerce/src/Plugin/Field/FieldFormatter/AttendeeFormatter.php [hunk 6]
```diff
@@ -66,15 +66,19 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
...
           '#url' => Url::fromRoute('dckyiv_commerce.attendee_edit', [
```
Comment (anpolimus) [inline docroot/modules/custom/dckyiv_commerce/src/Plugin/Field/FieldFormatter/AttendeeFormatter.php]: This route is deleted.

Comment (andriyun) [reply docroot/modules/custom/dckyiv_commerce/src/Plugin/Field/FieldFormatter/AttendeeFormatter.php]: This route is ok. Here is line where it's added https://github.com/drupal-ukraine/dcamp-site/pull/82/files#diff-6d8d937740d20646932a00e67ab56bc8R1
Probably you mixed up this route with 'dckyiv_commerce.attendee_add'.
Removed mention of 'add' route

**Comment by andriyun** [comment]
@anpolimus I've fixed changes. Please check again.

**Comment by anpolimus** [comment]
@andriyun , thank for your last updates.
Your changes are great.
remaining things:
- [ ] Change payment/complete page based on new feature;
- [ ] Change content of receipt email

**Comment by andriyun** [comment]
@anpolimus 
Could we move remaining things from your comment to follow up tasks and merge this PR?
There we can describe changes we missing.

**Comment by anpolimus** [comment]
1. @andriyun Важлива річ, яку треба зробити при створенні Attendee параграфа 
- треба з даних профіля FirstName Last Name прокинути ці дані в поля параграфа, який ми створюємо
2. Додати лінк на payment/complete сторінку
3. Зробити кастомний лист

**Comment by andriyun** [comment]
@anpolimus 
> треба з даних профіля FirstName Last Name прокинути ці дані в поля параграфа, який ми 
створюємо
Done
> Додати лінк на payment/complete сторінку
Done 
> Зробити кастомний лист
Done
+ Added styles for "my tickets" page


// @TODO
Render attendees info inside custom email.
I've implemented solution but view is empty at the current moment.

**Pull request merged**